Do some changes for `programming-lecture-artemis-score-stats`

Luro02 opened the pull request:
Note: Ignore the commit messages, I will squash/rewrite them soon

dfuchss commented:
Please add a short description what has changed why :) Otherwise, it's hard to determine whether a change (besides adding getters) makes sense :)

Feuermagier commented on src/main/java/edu/kit/kastel/sdq/artemis4j/client/ProgrammingSubmissionDTO.java:
```diff
@@ -51,9 +51,19 @@ public static void saveAssessment(ArtemisClient client, long participationId, bo
+            if (result.feedbacks() == null) {
+                result = result.fetchFeedbacks(client, this.participation().id());
+            }
+
```
Do you have an example where this is actually necessary (and we are not just using the wrong API endpoint for a certain task)?

Luro02 replied:
This is necessary, because of the current artemis4j implementation. It will always call the fetchlongfeedbacks for every result in a submission response. The method itself will remove all results that do not have feedback, so my submission will never have any results.

Without the results (the ids) I am unable to call the endpoint of the above code.

Luro02 replied:
The least intrusive option was to fetch the feedback when it is missing. It seems like the feedback is present under some conditions, so I would like to make Artemis4J handle the fetching of the feedback (if necessary). 

Similar to long feedbacks, the user doesn't want to manually fetch the feedbacks. They might not even expect them to be missing, like it happened to me. Took me a while to figure out that my results are all empty

Luro02 replied:
One idea that comes to mind, is to only fetch the long feedback/feedbacks if they are accessed. I assume the LazyNetworkValue could be used for that?

Feuermagier replied:
Ok, so it looks like the exercises/id/programming-submissions endpoint just never returns feedbacks. I guess the best solution would be to just remove long feedback fetching from ProgrammingSubmissionDTO::fetchAll, and implement a `fetchDetails(participationId, resultId)` method in `ResultDTO` (that fetches long feedbacks), similar to the old Artemis4J implementation.

Luro02 commented:
> Please add a short description what has changed why :) Otherwise, it's hard to determine whether a change (besides adding getters) makes sense :)

The diff is quite small, so I deemed it unnecessary to add a description. What changes should I describe? The getters are obvious and the only other change is the `openAssessment` method, for which I added Javadoc.

The `Assessment` is the abstraction over the `ResultDTO` of a submission. To see how a student has been graded, I need to open that assessment.

What might be mentioned: save/submit might not work, but if i remember correctly, Artemis lets you steal the assessment of another tutor.

Other than that, I adjusted the code to fetch feedbacks if they are missing and made the bug fix to the get relevant result, like we discussed on element.

dfuchss commented:
Could you open the PR in the repository itself? With your fork the CI cannot access the Secrets needed for the Test environment :)
